src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs

```diff
@@ -32,96 +32,6 @@ internal TypeofBinder(ExpressionSyntax typeExpression, Binder next)
-            private bool _seenConstructed;
-
-            /// <param name="typeSyntax">The argument to typeof.</param>
-            /// <param name="allowedMap">
-            /// Keys are GenericNameSyntax nodes representing unbound generic types.
-            /// Values are false if the node should result in an error and true otherwise.
-            /// </param>
-            public static void Visit(ExpressionSyntax typeSyntax, out Dictionary<GenericNameSyntax, bool> allowedMap)
-            {
-                OpenTypeVisitor visitor = new OpenTypeVisitor();
-                visitor.Visit(typeSyntax);
-                allowedMap = visitor._allowedMap;
-            }
-
-            public override void VisitGenericName(GenericNameSyntax node)
-            {
-                SeparatedSyntaxList<TypeSyntax> typeArguments = node.TypeArgumentList.Arguments;
-                if (node.IsUnboundGenericName)
-                {
-                    if (_allowedMap == null)
-                    {
-                        _allowedMap = new Dictionary<GenericNameSyntax, bool>();
-                    }
-                    _allowedMap[node] = !_seenConstructed;
-                }
-                else
-                {
-                    _seenConstructed = true;
-                    foreach (TypeSyntax arg in typeArguments)
-                    {
-                        Visit(arg);
-                    }
-                }
-            }
-
-            public override void VisitQualifiedName(QualifiedNameSyntax node)
-            {
-                bool seenConstructedBeforeRight = _seenConstructed;
-
-                // Visit Right first because it's smaller (to make backtracking cheaper).
-                Visit(node.Right);
-
-                bool seenConstructedBeforeLeft = _seenConstructed;
-
-                Visit(node.Left);
-
-                // If the first time we saw a constructed type was in Left, then we need to re-visit Right
-                if (!seenConstructedBeforeRight && !seenConstructedBeforeLeft && _seenConstructed)
-                {
-                    Visit(node.Right);
-                }
-            }
-
-            public override void VisitAliasQualifiedName(AliasQualifiedNameSyntax node)
-            {
-                Visit(node.Name);
-            }
-
-            public override void VisitArrayType(ArrayTypeSyntax node)
-            {
-                _seenConstructed = true;
-                Visit(node.ElementType);
-            }
-
-            public override void VisitPointerType(PointerTypeSyntax node)
-            {
-                _seenConstructed = true;
-                Visit(node.ElementType);
-            }
-
-            public override void VisitNullableType(NullableTypeSyntax node)
-            {
-                _seenConstructed = true;
-                Visit(node.ElementType);
-            }
-        }
+            => _allowedMap != null && _allowedMap.TryGetValue(syntax, out bool allowed) && allowed;
```

Comment:
>=> _allowedMap != null && _allowedMap.TryGetValue(syntax, out bool allowed) && allowed; [](http://example.com/codeflow?start=12&length=87)

Unless there is a desire to change behavior of the function, please revert the change